core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -163,7 +163,14 @@ func (p *logEventProvider) GetLogs(context.Context) ([]ocr2keepers.UpkeepPayload
 			p.lggr.Warnw("failed to pack log data", "err", err, "log", log)
 			continue
 		}
-		payload := ocr2keepers.NewUpkeepPayload(l.id, logTriggerType, ocr2keepers.BlockKey(fmt.Sprintf("%d", log.BlockNumber)), trig, checkData)
+
+		// TODO: we can't import evm package to here because of circular dependency
+		payload, err := NewUpkeepPayload(l.id, logTriggerType, trig, checkData)
```

Comment:
NewUpkeepPayload is in evm pkg, while this provider.go is in a different pkg. They have circular dependencies.